lib/src/tasks/saucelabs/cli.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright 2015 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library dart_dev.src.tasks.saucelabs.cli;
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:args/args.dart';
+
+import 'package:dart_dev/util.dart' show reporter, TaskProcess;
+
+import 'package:dart_dev/src/tasks/cli.dart';
+import 'package:dart_dev/src/tasks/config.dart';
+import 'package:dart_dev/src/tasks/saucelabs/api.dart';
+import 'package:dart_dev/src/tasks/saucelabs/sauce_runner.dart' as sauceRunner;
+import 'package:dart_dev/src/tasks/saucelabs/xml_reporter.dart';
+
+class SauceRunnerCli extends TaskCli {
+  final ArgParser argParser = new ArgParser()
+    ..addOption('build-name',
+        abbr: 'b', defaultsTo: 'saucelabs', help: 'Build name for the run.')
+    ..addFlag('new-sauce-tunnel',
+        defaultsTo: true,
+        negatable: true,
+        help: 'Create a new sauce-connect tunnel.')
+    ..addOption('sauce-tunnel-identifier',
+        abbr: 'i', help: 'Identifier to use in sauce tunnel creation.');
+
+  final String command = 'saucelabs';
+
+  final String sauceAccessKey = env['SAUCE_ACCESS_KEY'];
+  final String sauceUserName = env['SAUCE_USERNAME'];
+
+  Future<CliResult> run(ArgResults parsedArgs, {bool color: true}) async {
+    if (sauceUserName == null || sauceAccessKey == null) {
+      return new CliResult.fail('Sauce Labs credentials must be available via '
+          'the `SAUCE_ACCESS_KEY` and `SAUCE_USERNAME` SauceRunnerConfig instance.');
```

Comment:
This error message is confusing - it sounds like the key and username should be available on an instance of some Dart object, but it is actually looking for them from the shell environment via `dart:io`.